lib/scsi_defs.h

```diff
@@ -0,0 +1,84 @@
+#ifndef LIB_SCSI_DEFS_H
+#define LIB_SCSI_DEFS_H
+
+#include <cstdint>
+
+namespace scsi_defs {
+
+  // SAM-4 Table 33 
+  enum class Status : uint8_t {
+    kGood = 0x0,
+    kCheckCondition = 0x2,
+    kConditionMet = 0x4,
+    kBusy = 0x8,
+    kIntermediate = 0x10,              // obsolete
+    kIntermediateConditionMet = 0x14,  // obsolete
+    kReservationConflict = 0x18,
+    kCommandTerminated = 0x22,         // obsolete
+    kTaskSetFull = 0x28,
+    kAcaActive = 0x30,
+    kTaskAborted = 0x40,
+  };
+
+  // SCSI Reference Manual Table 61 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  enum class PeripheralDeviceType : uint8_t {
+    kDirectAccessBlock = 0x0,
+    kSequentialAccess = 0x1,
+    kPrinter = 0x2,
+    kProcessor = 0x3,
+    kWriteOnce = 0x4,
+    kCdDvd = 0x5,
+    kOpticalMemory = 0x7,
+    kMediumChanger = 0x8,
+    kStorageArrayController = 0xc,
+    kEnclosureServices = 0xd,
+    kSimplifiedDirectAccess = 0xe,
+    kOpticalCardReaderWriter = 0xf,
+    kBridgeControllerCommands = 0x10,
+    kObjectBasedStorage = 0x11,
+    kAutomationDriveInterface = 0x12,
+    kWellKnownLogicalUnit = 0x1e,
+    kUnknown = 0x1f,
+  };
+
+  // SCSI Reference Manual Table 62 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  // Field code of implemented version of the SPC standard 
+  enum class Version : uint8_t {
+    kNoStandard = 0x0,
+    kSpc = 0x3,
+    kSpc2 = 0x4,
+    kSpc3 = 0x5,
+    kSpc4 = 0x6,
+    kSpc5 = 0x7,
+  };
+
+  // SCSI Reference Manual https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  // Opt codes defined for SCSI commands supported by this project; this list may increase
+  enum class OptCode : uint8_t {
```

Comment:
Should be "opcodes", not "optcodes".  However, the scsi spec seems to prefer "commands".

Reply:
thanks for catching the "opcode"! Since this enum labels the operation code of each command and a "command" has other data fields in addition to operation code, I feel like it might make more sense to leave this as named OpCode. That being said, I'd be fine changing it to "commands" if that satisfies the specs better.

Reply:
Looks good to me like this.